tests/test_Source/test_supernovae_pop.py

```diff
@@ -0,0 +1,71 @@
+from slsim.Sources.Supernovae.supernovae_pop import (
+    calculate_star_formation_rate,
+    delay_time_distribution,
+)
+from slsim.Sources.Supernovae.supernovae_pop import SNIaRate
+from astropy.cosmology import FlatLambdaCDM
+import numpy as np
+import numpy.testing as npt
+
+cosmo = FlatLambdaCDM(70, 0.3)
+sne_rate = SNIaRate(cosmo, 10)
+
+
+def test_calculate_star_formation_rate():
+    z = 0
+    npt.assert_almost_equal(calculate_star_formation_rate(z), 0.0118, decimal=4)
+    z = 3
+    npt.assert_almost_equal(calculate_star_formation_rate(z), 0.1565, decimal=3)
+
+
+def test_delay_time_distribution():
+    t_d = -1
+    npt.assert_equal(delay_time_distribution(t_d), np.nan)
+    t_d = 5
+    npt.assert_almost_equal(delay_time_distribution(t_d), 5 ** (-1.08), decimal=4)
+
+
+class test_SNIaRate:
```

Comment:
classes need to start with capital TestSNIaRate() otherwise they are not being triggered.
Then I suggest you are using a 
def setup_method(self):
definition to set the one_rate and cosmo quantities as self variables within the class